server/storage/types.go

```diff
@@ -0,0 +1,446 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package storage
+
+import "github.com/CeresDB/ceresdbproto/pkg/clusterpb"
+
+type (
+	ClusterID    uint32
+	SchemaID     uint32
+	ShardID      uint32
+	TableID      uint64
+	ClusterState int
+)
+
+const (
+	Empty ClusterState = iota + 1
+	Stable
+)
+
+type ShardRole int
+
+const (
+	Leader ShardRole = iota + 1
+	Follower
+)
+
+type NodeState int
+
+const (
+	Online NodeState = iota + 1
+	Offline
+)
+
+type ListClustersResult struct {
+	Clusters []Cluster
+}
+
+type CreateClusterRequest struct {
+	Cluster Cluster
+}
+
+type CreateClusterViewRequest struct {
+	ClusterView ClusterView
+}
+
+type GetClusterViewRequest struct {
+	ClusterID ClusterID
+}
+
+type GetClusterViewResult struct {
+	ClusterView ClusterView
+}
+
+type PutClusterViewRequest struct {
+	ClusterID     ClusterID
+	ClusterView   ClusterView
+	LatestVersion uint64
+}
+
+type ListSchemasRequest struct {
+	ClusterID ClusterID
+}
+
+type ListSchemasResult struct {
+	Schemas []Schema
+}
+
+type CreateSchemaRequest struct {
+	ClusterID ClusterID
+	Schema    Schema
+}
+
+type CreateTableRequest struct {
+	ClusterID ClusterID
+	SchemaID  SchemaID
+	Table     Table
+}
+
+type GetTableRequest struct {
+	ClusterID ClusterID
+	SchemaID  SchemaID
+	TableName string
+}
+
+type GetTableResult struct {
+	Table  Table
+	Exists bool
```

Comment:
Why add this variable `Exists`? maybe we can just check table is exists like this:
```
GetTableResult.Table == nil
```

Reply:
`GetTableResult.Table` is a struct, can't use `GetTableResult.Table == nil`.